lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp

```diff
@@ -476,6 +476,7 @@ static void markDecomposedOpsAsIllegal(MLIRContext *context,
   target.addIllegalOp<PrimsSqueezeOp>();
   target.addIllegalOp<AtenMovedimIntOp>();
   target.addIllegalOp<AtenOneHotOp>();
+  target.addIllegalOp<AtenEinsumOp>();
```

Comment:
In DecomposeOps, we use `addPatternIfTargetOpIsIllegal`. I.e. the op will not be decomposed when the target declares it as legal. Wouldn't this conflict with declaring it illegal here?

Reply:
We set the ops back to legal if they are in the `backendLegalOpsSet`:

https://github.com/llvm/torch-mlir/blob/ae7f5775162bed34293392a11e366a97540cd65a/lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp#L485-L488

There's probably a clearer way of doing this.